tests/fixtures/MakeAuthenticatorLoginFormCustomUsernameField/src/Security/UserProvider.php

```diff
@@ -0,0 +1,60 @@
+<?php
+
+namespace App\Security;
+
+use Symfony\Component\Security\Core\Exception\UnsupportedUserException;
+use Symfony\Component\Security\Core\User\UserInterface;
+use Symfony\Component\Security\Core\User\UserProviderInterface;
+
+class UserProvider implements UserProviderInterface
+{
+    /**
+     * Symfony calls this method if you use features like switch_user
+     * or remember_me.
+     *
+     * If you're not using these features, you do not need to implement
+     * this method.
+     *
+     * @return UserInterface
+     */
+    public function loadUserByUsername($username)
+    {
+        return (new User())
+            ->setUserEmail($username)
+            ->setPassword($username);
+    }
+
+    /**
+     * Refreshes the user after being reloaded from the session.
+     *
+     * When a user is logged in, at the beginning of each request, the
+     * User object is loaded from the session and then this method is
+     * called. Your job is to make sure the user's data is still fresh by,
+     * for example, re-querying for fresh User data.
+     *
+     * If your firewall is "stateless: false" (for a pure API), this
+     * method is not called.
+     *
+     * @return UserInterface
+     */
+    public function refreshUser(UserInterface $user)
+    {
+        if (!$user instanceof User) {
+            throw new UnsupportedUserException(sprintf('Invalid user class "%s".', get_class($user)));
+        }
+
+        /* @var User $user */
+
+        // Return a User object after making sure its data is "fresh".
+        // Or throw a UsernameNotFoundException if the user no longer exists.
+        throw new \Exception('TODO: fill in refreshUser() inside '.__FILE__);
+    }
+
+    /**
+     * Tells Symfony to use this provider for this User class.
+     */
+    public function supportsClass($class)
+    {
+        return User::class === $class;
+    }
+}
```

Comment:
Nit pick - but could we "slim down" some of these fixture classes, so that they only have what we need?